### ISSUE TITLE
Collect IP addresses from new volunteers to exclude from Google Analytics

### ISSUE THREAD

**Issue by sheltoG** [hackforla/website#2368]
### Dependency
Launch homepage and revisit the practically

### Overview
We need to exclude the IP addresses of volunteers from Google Analytics to filter the traffic volunteers generate in visiting the website from normal users .

### Action Items
- [ ] Research a way to get the public ip address of a volunteer
    - [ ] Most likely this will implement some API from a third party service like Cloudflare to get the IP
    - [ ] We can have a button (or some kind of link) hidden in the website that will record the IP of a volunteer
- [ ] Store this data in our google admin drive
    - [ ] Because of the nature of IP addresses, this information should be hidden and secured
    - [ ] The initial ask was to gather the first three octets of the user's IP (i.e. 10.100.100.*); however, this will exclude the entire subnet, which means if a volunteer's public IP is in the same subnet of a potential valid user, then the valid user's data will be lost
    - [ ] The data can be set to expire and be discarded in 180 days
- [ ] Upload this data as an excluded list into google analytics
    - [ ] Most likely will need some kind of API from google

**Comment by macho-catt** [comment]
@ExperimentsInHonesty I just thought of this after the meeting ended, but if we're only getting the first three octets of the public IP (i.e. 10.100.100.xxx), then all 254 possible clients in that octet will be excluded from google analytics. That means if, for example, my public IP was 10.100.100.02 and we exclude 10.100.100.xxx, then valid users with IPs ranging from 10.100.100.03 until 10.100.100.254 will be excluded from the list as well, and we won't get their traffic. Is that ok?

Also something fun to think about. ISPs will generally give the same area the same public IP address. So the public ip address that I have will most likely be the same as my neighbor's, if we are using the same ISP provider. :D

**Comment by ExperimentsInHonesty** [comment]
@macho-catt Yeah, after we spoke I found this https://help.analyticsedge.com/article/filter-working-from-home-employees-in-google-analytics/ which gives a way to exclude users but as it points out in the article, that excludes them retroactively, so given that our developers are people who will have visited the site and converted to members, that would artificially depress our numbers.  Its really sessions we want to exclude.  Not sure how to do that.

**Comment by ExperimentsInHonesty** [comment]
Here is another article that is worth talking about https://mixedanalytics.com/blog/exclude-self-google-analytics-no-fixed-ip/

**Comment by macho-catt** [comment]
Looks like there are other options too, based on these two articles I read:

https://www.daniloaz.com/en/5-ways-to-exclude-your-own-visits-from-google-analytics/
https://www.analyticsmania.com/post/how-to-exclude-internal-traffic-in-google-analytics-4/

So IP filtering and using a third party browser extension are mentioned as well, but to add to those we can either use:

1) Cookies on each device
- **Pros**
    - GA excludes based on sessions, and not IP
    - Relatively simple to implement
- **Cons**
    - Cookies will need to be set up on each device that you want to exclude
    - If cookies are cleared, they need to be put back
2) A variable can be put on the webpage, so if a dev goes to the url with that specific variable (i.e. hackforla.org/?test=1) we ignore the session
- **Pros**
    - Simpler to implement
    - GA excludes based on sessions, and not IP
    - A dev can choose if they want to go to the non-tracked page
- **Cons**
    - A dev will need to remember to go to the right page
